common-util/Details/ServiceState/3rdStepFinishedRegistration/utils.jsx

```diff
@@ -159,39 +165,45 @@ export const handleMultisigSubmit = async ({
               multisigContractWeb3.methods
                 .approveHash(messageHash)
                 .send({ from: account })
-                .on('transactionHash', async (hash) => {
-                  const transactionDetails = await poll(hash, chainId);
-                  console.log({ transactionDetails });
-
-                  // Get the signature bytes based on the account address, since it had its tx pre-approved
-                  const signatureBytes = `0x000000000000000000000000${account.slice(
-                    2,
-                  )}0000000000000000000000000000000000000000000000000000000000000000`
-                    + '01';
-
-                  const safeExecData = multisigContract.interface.encodeFunctionData(
-                    'execTransaction',
-                    [
-                      safeTx.to,
-                      safeTx.value,
-                      safeTx.data,
-                      safeTx.operation,
-                      safeTx.safeTxGas,
-                      safeTx.baseGas,
-                      safeTx.gasPrice,
-                      safeTx.gasToken,
-                      safeTx.refundReceiver,
-                      signatureBytes,
-                    ],
-                  );
-
-                  // Redeploy the service updating the multisig with new owners and threshold
-                  const packedData = ethers.utils.solidityPack(
-                    ['address', 'bytes'],
-                    [multisig, safeExecData],
-                  );
-
-                  handleStep3Deploy(radioValue, packedData);
+                .on('transactionHash', async () => {
+                    await multisigContractWeb3.getPastEvents('ApproveHash', {
+                            fromBlock: startingBlock - 10,
+                            toBlock: 'latest'
+                            },
+                        (error, events) => {
+                          console.log("all events", events);
+                          if (events.length > 0) {
```

Comment:
@mohandast52 here we would like to poll on past events of our transaction and wait until we catch the one with the latest 10 blocks. Is this possible somehow?